### PR TITLE
feat/1248 - Search by Ids

### PR DESCRIPTION
**Pull request by jberube**

### CODE REVIEW

File: src/components/CohortBuilder/Categories/ActionCategory.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { noop } from 'lodash';
```
Comment (evans-g-crsj) [inline src/components/CohortBuilder/Categories/ActionCategory.js]: Following our discussion on lodash vs tree-shaking : wouldn't it be more appropriate to do import noop from 'lodash/noop'; ?

File: src/components/CohortBuilder/Categories/ActionCategory.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+import Column from 'uikit/Column';
+
+export default class ActionCategory extends React.PureComponent {
```
Comment (evans-g-crsj) [inline src/components/CohortBuilder/Categories/ActionCategory.js]: Should be a stateless component ?

File: src/components/CohortBuilder/SearchById/SearchByIdModal.js [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+      .catch(console.err);
+  }
+
+  handleInputIdsChange(evt) {
+    const inputIdsText = evt.currentTarget.value;
+    this.setInputText(inputIdsText);
+  }
+
+  async handleViewResultsClick() {
+    if (this.state.loading === true) return;
```
Comment (evans-g-crsj) [inline src/components/CohortBuilder/SearchById/SearchByIdModal.js]: can " if (this.state.loading) " do it?

File: src/components/CohortBuilder/SearchById/SearchByIdModal.js [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+          Apply Filter
+        </TealActionButton>
+      </React.Fragment>
+    );
+  }
+
+  render() {
+    return (
+      <React.Fragment>
+        <Row className="header">{this.renderHeader()}</Row>
```
Comment (evans-g-crsj) [inline src/components/CohortBuilder/SearchById/SearchByIdModal.js]: It s really a matter of personal choice but I think I would make a component for each render (renderHeader, renderFooter,...) methods

File: src/uikit/DataTable/ControlledDataTable.js [hunk 5]
```diff
@@ -1,46 +1,74 @@
...
+      className = '',
       striped = true,
+      style = {},
+
+      // pagination-related stuff
+      showPagination = true,
+      manualPagination = true,
+      defaultPageSize = 20,
+      dataTotalCount = -1,
+      onFetchData = () => null,
```
Comment (evans-g-crsj) [inline src/uikit/DataTable/ControlledDataTable.js]: should we use noop from lodash?

File: src/components/CohortBuilder/Categories/ActionCategory.js (hunk 1 above)
Comment (jberube) [reply src/components/CohortBuilder/Categories/ActionCategory.js]: Yes, but it only pays if it's done everywhere, so it's not an issue to leave it this way for the moment.

File: src/components/CohortBuilder/Categories/ActionCategory.js (hunk 2 above)
Comment (jberube) [reply src/components/CohortBuilder/Categories/ActionCategory.js]: It is. Inheriting `React.PureComponent` makes it a stateless component.

File: src/components/CohortBuilder/Categories/ActionCategory.js (hunk 1 above)
Comment (jberube) [reply src/components/CohortBuilder/Categories/ActionCategory.js]: Bah, done it anyway :p

File: src/components/CohortBuilder/SearchById/SearchByIdModal.js (hunk 3 above)
Comment (jberube) [reply src/components/CohortBuilder/SearchById/SearchByIdModal.js]: Yes!

**Pull request merged**